source/MaterialXCore/Document.cpp

```diff
@@ -1094,8 +1094,36 @@ void Document::upgradeVersion(bool applyFutureUpdates)
                     node->setName(newNodeName);
                 }
             }
+        }   
+
+        // Convert all parameters to be inputs. If needed set them to be "uniform".
+        for (ElementPtr e : traverseTree())
```

Comment:
Simple update traversal code. Note:
1) There is no change in categorization of existing <Parameters>. They are all tagged as being uniform. Then can be selectively not be marked as uniform according to spec rules.
2) The ordering of children is being maintained by adding and them moving the new <Input> child to where the previous <Parameter> child existed.